daemon/cluster/networks.go

```diff
@@ -83,13 +83,21 @@ func attacherKey(target, containerID string) string {
 func (c *Cluster) UpdateAttachment(target, containerID string, config *network.NetworkingConfig) error {
 	c.mu.RLock()
 	attacher, ok := c.attachers[attacherKey(target, containerID)]
-	c.mu.RUnlock()
 	if !ok || attacher == nil {
+		c.mu.RUnlock()
 		return fmt.Errorf("could not find attacher for container %s to network %s", containerID, target)
 	}
+	if attacher.inProgress {
+		logrus.Debugf("Discarding redundant notice of resource allocation on network %s for task id %s",
+			target, attacher.taskID)
+		c.mu.RUnlock()
+		return nil
+	}
+	attacher.inProgress = true
```

Comment:
Good catch, did not realize it was just using a read lock.
Now that it is doing this new setting, it makes sense to change it to a write lock.